App.js

```diff
@@ -161,73 +322,311 @@ export default function App() {
+      if (prevGameState.whoseTurn === 2 && workBoard[newPositionBoard2].extraScore !== undefined) {
+        scoreAdj = scoreAdj + workBoard[newPositionBoard2].extraScore;
+        if (workBoard[newPositionBoard2].extraScore < 0) {
+          if (prevGameState.whoseTurn === 1 && prevGameState.numberOfPlayers === 2) {
+            workWhoseTurn = calculateTurn(workTurnTotals,1,-1);
+            if (workMessage === 'Great...roll doubled') {
+              workMessage = 'Roll Doubled (extra roll)';
+            } else {
+              workMessage = '(Player 1 extra roll)';
+            }
+            workOptMessage = 'Roll Player 1';
+          } else if (prevGameState.whoseTurn === 2 && prevGameState.numberOfPlayers === 2) {
+            workWhoseTurn = calculateTurn(workTurnTotals,-1,1);
+            if (workMessage === 'Great...roll doubled') {
+              workMessage = 'Roll Doubled (extra roll)';
+            } else {
+              workMessage = '(Player 2 extra roll)';
+            }
+            workOptMessage = 'Roll Player 2';
+          } else {
+            workWhoseTurn = calculateTurn(workTurnTotals,1,-1);
+            if (workMessage === 'Great...roll doubled') {
+              workMessage = 'Roll Doubled (extra roll)';
+            } else {
+              workMessage = '(extra roll)';
+            }
+            workOptMessage = 'Roll Player 1';
+          }
+        } else {
+          if (prevGameState.whoseTurn === 1 && prevGameState.numberOfPlayers === 2) {
+            workWhoseTurn = calculateTurn(workTurnTotals,-1,1);
+            if (workMessage === 'Great...roll doubled') {
+              workMessage = 'Roll Doubled (lose a roll)';
+            } else {
+              workMessage = '(Player 1 lose a roll)';
+            }
+            workOptMessage = 'Roll Player 2';
+          } else if (prevGameState.whoseTurn === 2 && prevGameState.numberOfPlayers === 2) {
+            workWhoseTurn = calculateTurn(workTurnTotals,1,-1);
+            if (workMessage === 'Great...roll doubled') {
+              workMessage = 'Roll Doubled (lose a roll)';
+            } else {
+              workMessage = '(Player 2 lose a roll)';
+            }
+            workOptMessage = 'Roll Player 1';
+          } else {
+            workWhoseTurn = calculateTurn(workTurnTotals,1,-1);
+            if (workMessage === 'Great...roll doubled') {
+              workMessage = 'Roll Doubled (lose a roll)';
+            } else {
+              workMessage = '(lose a roll)';
+            }
+            workOptMessage = 'Roll Player 1';
+          }
         }
       }
       // Add the detour squares and remove a single square after detour
-      if (workBoard[newPositionBoard1].itemsToAdd !== undefined) {
+      // Player 1 landed on detour spot
+      if (workBoard[newPositionBoard1].itemsToAdd !== undefined && prevGameState.whoseTurn === 1) {
         let workBoard2 = workBoard.slice();
+        let detourPriorlyOpened = false;
         let i;
```

Comment:
It seems like you are using this `i` term for different things that makes it confusing :S
So it starts as your loop iterating, so `i` kinda makes sense....but then you use it beyond your loop? so now it's just workBarod2.length - 1? It would probably be clearer if you just used `i` within the array and then used workBarod2.length - 1 outside